frontend/model/contracts/group.js

```diff
@@ -1889,27 +1825,33 @@ sbp('chelonia/defineContract', {
           })
           return
         }
-      } */
+      }
 
       leaveAllChatRoomsUponLeaving(state, memberID, innerSigningContractID).catch((e) => {
         console.warn('[gi.contracts/group/leaveGroup]: Error while leaving all chatrooms', e)
       })
 
       if (memberID === identityContractID) {
         const possiblyUselessContractIDs = Object.keys(state.profiles || {}).filter(cID => cID !== identityContractID)
-        sbp('gi.actions/group/removeUselessIdentityContracts', {
-          contractID,
-          possiblyUselessContractIDs
-        })
+        sbp('chelonia/contract/release', possiblyUselessContractIDs).catch(e =>
+          console.error('[gi.contracts/group/leaveGroup] Error calling release on all members', e)
+        )
 
-        // we can't await on this in here, because it will cause a deadlock, since Chelonia processes
-        // this method on the eventqueue for this contractID, and /remove uses that same eventqueue
-        sbp('chelonia/contract/remove', contractID).catch(e => {
-          console.error(`sideEffect(removeMember): ${e.name} thrown by /remove ${contractID}:`, e)
+        sbp('gi.actions/identity/leaveGroup', {
+          contractID: identityContractID,
+          data: {
+            groupContractID: contractID
+          }
+        }).catch(e => {
+          console.warn(`[gi.contracts/group/leaveGroup] ${e.name} thrown by gi.contracts/identity/leaveGroup ${identityContractID} for ${contractID}:`, e)
         })
       } else {
         const myProfile = getters.groupProfile(identityContractID)
 
+        // Do _not_ release memberID, because doing so will remove their profile
+        // from our view
+        // // sbp('chelonia/contract/release', memberID)
+
```

Comment:
Maybe a comment can be added here as a `// TODO` (and a new issue) to mention that we could simply save in a local cache their username + profile picture and load that in the UI instead of remaining subscribed to their contract.